fortex/health/processors/scispacy_processor.py

```diff
@@ -0,0 +1,163 @@
+                    tmp_abrv = Abbreviation(
+                        pack=input_pack, begin=abrv.start, end=abrv.end
+                    )
+                    tmp_abrv.long_form = abrv._.long_form
+                    list_of_abrvs.append(tmp_abrv)
+
+            else:
+                for item in doc._.hearst_patterns:
+                    general_concept: Phrase = Phrase(
+                        pack=input_pack, begin=item[1].start, end=item[1].end
+                    )
+                    specific_concept = Phrase(
+                        pack=input_pack, begin=item[2].start, end=item[2].end
+                    )
+                    hlink = Hyponym(
+                        pack=input_pack,
+                        parent=general_concept,
+                        child=specific_concept,
+                    )
+                    hlink.hyponym_link = item[0]
+
+    @classmethod
+    def default_configs(cls):
+        r"""
+        This defines a basic config structure for `ScispaCyProcessor`.
+
+        Following are the keys for this dictionary:
+         - `entry_type`: input entry type for classification
+         - `model_name`: the ScispaCy model name to be
+                         used for classification, default to
+                         en_core_sci_sm
+         - `pipe_name`: the Spacy model pipe name for
+                         classification, only 2 options here:
+                         abbreviation_detector or hyponym_detector
+
+        Returns: A dictionary with the default config for this processor.
+        """
+        return {
+            "entry_type": "ft.onto.base_ontology.Document",
+            "model_name": "en_core_sci_sm",
+            "pipe_name": "abbreviation_detector",
+        }
+
+    def expected_types_and_attributes(self):
+        r"""
+        Method to add user specified expected type which would be checked
+        before running the processor if the pipeline is initialized with
+        `enforce_consistency=True` or
+        :meth:`~forte.pipeline.Pipeline.enforce_consistency` was enabled for
+        the pipeline.
+        """
+        return {
+            self.configs.entry_type: set(),
+        }
+
+    def record(self, record_meta: Dict[str, Set[str]]):
+        r"""
+        Method to add output type record of `ScispaCyProcessor` which
+        is `"ftx.medical.clinical_ontology.hyponym"` with attributes:
+         `hyponym_link`
+        to :attr:`forte.data.data_pack.Meta.record`.
+
+        Args:
+            record_meta: the field in the datapack for type record that need to
+                fill in for consistency checking.
+        """
+        record_meta["ftx.medical.clinical_ontology.Abbreviation"] = {
+            "long_form",
+        }
+        record_meta["ftx.medical.clinical_ontology.Hyponym"] = {
+            "hyponym_link",
+            "parent",
+            "child",
+        }
+
+        if self.configs.entry_type in record_meta:
```

Comment:
The following lines are problematic, we don't even have `self.configs.attribute_name`

Reply:
Ok this is now fixed (after referencing to style/example of negation_context_analyzer) and also "Phrase" is added as suggested.